### DISCUSSION TITLE
Truffle installation failure for Windows 10

### DISCUSSION THREAD

**Discussion by Swarup9873** [ConsenSys-archive/truffle#5248] [Installation]
![cmd truffle](https://user-images.githubusercontent.com/97334699/176596894-72124d2c-8735-42d3-8775-aec1f8555f1a.PNG)

Hello everyone, I am facing these errors while installing Truffle.
Plz guide me through the steps that I should take in order to successfully install Truffle in my Windows 10.

**Comment by apdan510** [comment]
@Swarup9873 
This error is saying that there is no visual studio compiler is installed on your computer, please follow these steps to install it:

**Open an administrator CMD line or Powershell:**
**Make sure Chocolatey is installed by pasting the below command into an administrative Cmdline or Powershell:**
`Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://community.chocolatey.org/install.ps1'))`

**Install Visual studio 2019 build tools**
`choco install -y visualstudio2019buildtools visualstudio2019-workload-vctools`

**Set VS studio 2019 build tools**
`npm config set msvs_version 2019`

**Then try installing Truffle once more:**
`npm install -g truffle`

Hope this helps!
Alex | Truffle Support | Consensys
